### PR TITLE
Add spring boot 2 integration module.

### PR DESCRIPTION
**Pull request by danielimre**
### :pencil: Description
Adds spring boot 2 integration module.
Adds integration tests for spring boot 2 with webflux, tracing, and prometheus metrics.
Fixes deprecated and removed reactor-netty api usages.

### CODE REVIEW

**Review by gaborsmith — APPROVED** [review]

File: molten-spring-boot/src/main/java/com/hotels/molten/spring/boot/MoltenCoreSpringBootInitializer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * Initializes Molten for spring-boot.
+ * Besides core and MDC initialization it also enables dimensional metrics.
+ * Registers MDC propagation for each operator.
+ */
+public class MoltenCoreSpringBootInitializer implements ApplicationListener<ApplicationEnvironmentPreparedEvent> {
+    @Override
+    public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event) {
+        MoltenCore.initialize();
+        MoltenMDC.initialize(false);
+        MoltenMetrics.setDimensionalMetricsEnabled(true);
```
Comment (adambriny) [inline molten-spring-boot/src/main/java/com/hotels/molten/spring/boot/MoltenCoreSpringBootInitializer.java:35]: Sorry if it's trivial, but why don't we need to initialize `MoltenTrace`?

File: molten-core/src/main/java/com/hotels/molten/core/mdc/MDCContextPropagatingSubscriber.java [hunk 2]
```diff
@@ -30,12 +30,12 @@
...
 @Slf4j
 final class MDCContextPropagatingSubscriber<T> extends AbstractNonFusingSubscription<T> {
-    private static final String MDC_CONTEXT_KEY = "X-MDC";
+    private static final String MDC_CONTEXT_KEY = "X-Molten-MDC";
     private final Context context;
 
     private MDCContextPropagatingSubscriber(CoreSubscriber<? super T> subscriber, Map<String, String> mdc) {
         super(subscriber);
-        this.context = subscriber.currentContext().put(MDC_CONTEXT_KEY, mdc);
+        this.context = mdc != null ? subscriber.currentContext().put(MDC_CONTEXT_KEY, mdc) : subscriber.currentContext();
```
Comment (adambriny) [inline molten-core/src/main/java/com/hotels/molten/core/mdc/MDCContextPropagatingSubscriber.java:38]: Mdc nullability is already checked at line 45 in the `decorate(...)` method.

File: molten-trace/src/main/java/com/hotels/molten/trace/MoltenTrace.java [hunk 3]
```diff
@@ -56,7 +56,7 @@ public static void initialize() {
      */
     public static void initialize(boolean onEachOperator) {
         uninitialize();
-        LOG.info("Integrating tracing with Molten...");
+        LOG.info("Integrating tracing with Molten onEachOperator={}", onEachOperator);
```
Comment (adambriny) [inline molten-trace/src/main/java/com/hotels/molten/trace/MoltenTrace.java:59]: Please consider adding this to the other `Molten*.initialize(boolean onEachOperator)` methods

Comment (danielimre) [reply molten-trace/src/main/java/com/hotels/molten/trace/MoltenTrace.java:59]: Done in fd3e6c3

File: molten-core/src/main/java/com/hotels/molten/core/mdc/MDCContextPropagatingSubscriber.java (hunk 2 above)
Comment (danielimre) [reply molten-core/src/main/java/com/hotels/molten/core/mdc/MDCContextPropagatingSubscriber.java]: Removed check in fd3e6c3

File: molten-trace-test/src/test/java/com/hotels/molten/trace/MoltenTraceTest.java [hunk 4]
```diff
@@ -74,6 +91,43 @@ public void should_support_nesting(boolean onEachOperatorEnabled) {
...
+        Assertions.assertThat(capturedSpans())
+            .anySatisfy(span -> {
+                Assertions.assertThat(span).extracting(Span::parentId).isEqualTo(outerSpan.id());
+                Assertions.assertThat(span).extracting(Span::name).isEqualTo("mid");
+            });
+        Assertions.assertThat(capturedSpans())
+            .anySatisfy(span -> {
+                Assertions.assertThat(span).extracting(Span::parentId).isEqualTo(midSpan.id());
+                Assertions.assertThat(span).extracting(Span::name).isEqualTo("inner");
+            });
```
[the remark this answers is not in the document]
Comment (danielimre) [reply molten-trace-test/src/test/java/com/hotels/molten/trace/MoltenTraceTest.java]: Thanks. Reorganized in fd3e6c3

File: molten-trace/src/main/java/com/hotels/molten/trace/TracingTransformer.java [hunk 5]
```diff
@@ -28,6 +28,8 @@
  * Transformer to wrap upstream in a trace span. The span is opened when upstream is subscribed to and closed when it completed, ran to an error, or was cancelled.
  * <p>
  * <strong>Be sure to have tracing initialized with {@link MoltenTrace#initialize()}.</strong>
+ * <strong>Please note that this solution not necessarily yields the expected result and should be handled with care.
+ * If the thread is changed between subscription and completion the span won't be nested and closed properly.</strong>
```
Comment (adambriny) [inline molten-trace/src/main/java/com/hotels/molten/trace/TracingTransformer.java:32]: Could you please create an issue for that?

**Review by adambriny — APPROVED** [review]

File: molten-trace/src/main/java/com/hotels/molten/trace/TracingTransformer.java (hunk 5 above)
Comment (danielimre) [reply molten-trace/src/main/java/com/hotels/molten/trace/TracingTransformer.java:32]: Created #36

**Review by gaborsmith — APPROVED** [review]

**Pull request merged**